### DISCUSSION TITLE
How to specify host & path for frontend websocket ?

### DISCUSSION THREAD

**Discussion by sesky4** [eclipse-theia/theia#11904] [Q&A] [answered]
I'm trying to `serve theia frontend file(html/js/css) in site1.com` and `hosting backend in site2.com`, how can I make fontend connect to ws://site2.com:1234/my_path ?

I looked into source code and found that the websocket address seems bound to current location, is there any way to work around with it ? ( without code modification if possible)
```
class WebSocketConnectionProvider {
    protected createWebSocketUrl(path: string): string {
        return new Endpoint().getWebSocketUrl().withPath(path).toString();
    }
}

class Endpoint {
    constructor(
        protected readonly options: Endpoint.Options = {},
        protected readonly location: Endpoint.Location = self.location
    ) { }
}
```

And thanks for you guys awesome work ! :D

**Comment by msujew** [comment, marked as answer]
Hey @sesky4,

you can always override the `WebSocketConnectionProvider` class and rebind it to your own implementation. I.e. an implementation that overrides the `createWebSocketUrl` method and points it towards your `site2.com` endpoint:

```ts
rebind(WebSocketConnectionProvider).to(RedirectingWebSocketConnectionProvider).inSingletonScope();
```

**Reply by sesky4** [reply]
Thanks ❤️
